### PR TITLE
Put back some deprecated for a bit longer and backward compat for loadDFG

### LINKED ISSUE #618: Compatibility of loadDFG! with new module naming?
HI @Affie , is it reasonable to require compatibility with new module names inside serialization?  For example a FileDFG object that does not have any module information should be able to load like it always has?

Use this as a test: https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/issues/582#issuecomment-671776919  which should load and give something similar to https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/issues/582#issuecomment-671890696

```julia
julia> fg = loadDFG("tenSolves.tar.gz")
sfolder = split(dstname, '.') = SubString{String}["tenSolves", "tar", "gz"]
[ Info: loadDFG! detected a gzip tenSolves.tar.gz -- unpacking via /tmp/caesar/random/94379f92 now...
┌ Error: Unable to deserialize soft type Pose2()
└ @ DistributedFactorGraphs ~/.julia/dev/DistributedFactorGraphs/src/services/Serialization.jl:70
┌ Error: UndefVarError: Pose2() not defined
│ Stacktrace:
│  [1] getTypeFromSerializationModule(::String) at /home/dehann/.julia/dev/DistributedFactorGraphs/src/services/Serialization.jl:67
│  [2] unpackVariableNodeData(::LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor}, ::DistributedFactorGraphs.PackedVariableNodeData) at /home/dehann/.julia/dev/DistributedFactorGraphs/src/services/Serialization.jl:193
│  [3] #86 at /home/dehann/.julia/dev/DistributedFactorGraphs/src/services/Serialization.jl:125 [inlined]
│  [4] iterate at ./generator.jl:47 [inlined]
│  [5] collect(::Base.Generator{Base.ValueIterator{Dict{String,DistributedFactorGraphs.PackedVariableNodeData}},DistributedFactorGraphs.var"#86#87"{LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor}}}) at ./array.jl:686
│  [6] map at ./abstractarray.jl:2188 [inlined]
│  [7] unpackVariable(::LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor}, ::Dict{String,Any}; unpackPPEs::Bool, unpackSolverData::Bool, unpackBigData::Bool) at /home/dehann/.julia/dev/DistributedFactorGraphs/src/services/Serialization.jl:125
│  [8] unpackVariable at /home/dehann/.julia/dev/DistributedFactorGraphs/src/services/Serialization.jl:103 [inlined]
│  [9] (::DistributedFactorGraphs.var"#120#126"{LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor},Array{DistributedFactorGraphs.DFGVariable,1}})(::IOStream) at /home/dehann/.julia/dev/DistributedFactorGraphs/src/FileDFG/services/FileDFG.jl:135
│  [10] open(::DistributedFactorGraphs.var"#120#126"{LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor},Array{DistributedFactorGraphs.DFGVariable,1}}, ::String; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at ./io.jl:325
│  [11] open at ./io.jl:323 [inlined]
│  [12] loadDFG!(::LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor}, ::String) at /home/dehann/.julia/dev/DistributedFactorGraphs/src/FileDFG/services/FileDFG.jl:133
│  [13] loadDFG(::String) at /home/dehann/.julia/dev/IncrementalInference/src/FGOSUtils.jl:358
│  [14] top-level scope at REPL[3]:1
│  [15] eval(::Module, ::Any) at ./boot.jl:331
│  [16] eval_user_input(::Any, ::REPL.REPLBackend) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:134
│  [17] repl_backend_loop(::REPL.REPLBackend) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:195
│  [18] start_repl_backend(::REPL.REPLBackend, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:180
│  [19] run_repl(::REPL.AbstractREPL, ::Any; backend_on_current_task::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:292
│  [20] run_repl(::REPL.AbstractREPL, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:288
│  [21] (::Base.var"#806#808"{Bool,Bool,Bool,Bool})(::Module) at ./client.jl:399
│  [22] #invokelatest#1 at ./essentials.jl:710 [inlined]
│  [23] invokelatest at ./essentials.jl:709 [inlined]
│  [24] run_main_repl(::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at ./client.jl:383
│  [25] exec_options(::Base.JLOptions) at ./client.jl:313
│  [26] _start() at ./client.jl:506
└ @ DistributedFactorGraphs ~/.julia/dev/DistributedFactorGraphs/src/services/Serialization.jl:74
ERROR: The variable doesn't seem to have a softtype. It needs to set up with an InferenceVariable from IIF. This will happen if you use DFG to add serialized variables directly and try use them. Please use IncrementalInference.addVariable().
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] unpackVariableNodeData(::LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor}, ::DistributedFactorGraphs.PackedVariableNodeData) at /home/dehann/.julia/dev/DistributedFactorGraphs/src/services/Serialization.jl:194
 [3] #86 at /home/dehann/.julia/dev/DistributedFactorGraphs/src/services/Serialization.jl:125 [inlined]
 [4] iterate at ./generator.jl:47 [inlined]
 [5] collect(::Base.Generator{Base.ValueIterator{Dict{String,DistributedFactorGraphs.PackedVariableNodeData}},DistributedFactorGraphs.var"#86#87"{LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor}}}) at ./array.jl:686
 [6] map at ./abstractarray.jl:2188 [inlined]
 [7] unpackVariable(::LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor}, ::Dict{String,Any}; unpackPPEs::Bool, unpackSolverData::Bool, unpackBigData::Bool) at /home/dehann/.julia/dev/DistributedFactorGraphs/src/services/Serialization.jl:125
 [8] unpackVariable at /home/dehann/.julia/dev/DistributedFactorGraphs/src/services/Serialization.jl:103 [inlined]
 [9] (::DistributedFactorGraphs.var"#120#126"{LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor},Array{DistributedFactorGraphs.DFGVariable,1}})(::IOStream) at /home/dehann/.julia/dev/DistributedFactorGraphs/src/FileDFG/services/FileDFG.jl:135
 [10] open(::DistributedFactorGraphs.var"#120#126"{LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor},Array{DistributedFactorGraphs.DFGVariable,1}}, ::String; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at ./io.jl:325
 [11] open at ./io.jl:323 [inlined]
 [12] loadDFG!(::LightDFG{SolverParams,DistributedFactorGraphs.DFGVariable,DistributedFactorGraphs.DFGFactor}, ::String) at /home/dehann/.julia/dev/DistributedFactorGraphs/src/FileDFG/services/FileDFG.jl:133
 [13] loadDFG(::String) at /home/dehann/.julia/dev/IncrementalInference/src/FGOSUtils.jl:358
 [14] top-level scope at REPL[3]:1
```

### PR DESCRIPTION
**Pull request by Affie**
should close #618

### CODE REVIEW

File: src/Deprecated.jl [hunk 1]
```diff
@@ -1,13 +1,57 @@
...
+
+# These will become AbstractPrior, AbstractRelativeFactor, and AbstractRelativeFactorMinimize in 0.9.
+abstract type FunctorSingleton <: FunctorInferenceType end
+abstract type FunctorPairwise <: FunctorInferenceType end
+abstract type FunctorPairwiseMinimize <: FunctorInferenceType end
+
+# I don't know how to deprecate this, any suggestions?
+const AbstractBigDataEntry = AbstractDataEntry
+
+@deprecate GeneralBigDataEntry(args...; kwargs...) GeneralDataEntry(args...; kwargs...)
```
Comment (Affie) [inline src/Deprecated.jl:32]: I would say we can remove all the "BigData" reference already.

File: src/Deprecated.jl [hunk 2]
```diff
@@ -1,13 +1,57 @@
...
+
+export InferenceType
+export FunctorSingleton, FunctorPairwise, FunctorPairwiseMinimize
+
+abstract type InferenceType end
+
+# These will become AbstractPrior, AbstractRelativeFactor, and AbstractRelativeFactorMinimize in 0.9.
+abstract type FunctorSingleton <: FunctorInferenceType end
+abstract type FunctorPairwise <: FunctorInferenceType end
+abstract type FunctorPairwiseMinimize <: FunctorInferenceType end
```
Comment (Affie) [inline src/Deprecated.jl:27]: @dehann If you can check if these are ok to remove now.

File: src/Deprecated.jl [hunk 3]
```diff
@@ -1,13 +1,57 @@
...
+##==============================================================================
+
+# temporary promote with warning
+Base.promote_rule(::Type{DateTime}, ::Type{ZonedDateTime}) = DateTime
+function Base.convert(::Type{DateTime}, ts::ZonedDateTime)
+    @warn "DFG now uses ZonedDateTime, temporary promoting and converting to DateTime local time"
+    return DateTime(ts, Local)
+end
+
+@deprecate listSolvekeys(x...) listSolveKeys(x...)
```
Comment (Affie) [inline src/Deprecated.jl:15]: Keep these.

File: src/Deprecated.jl [hunk 4]
```diff
@@ -1,13 +1,57 @@
...
+@deprecate listStoreEntries(args...) listDataBlobs(args...)
+@deprecate hasBigDataEntry(args...) hasDataEntry(args...)
+
+@deprecate getBigDataEntry(args...) getDataEntry(args...)
+@deprecate addBigDataEntry!(args...)  addDataEntry!(args...)
+@deprecate updateBigDataEntry!(args...) updateDataEntry!(args...)
+@deprecate deleteBigDataEntry!(args...) deleteDataEntry!(args...)
+@deprecate getBigDataKeys(args...) listDataEntries(args...)
+@deprecate getBigDataEntries(args...) getDataEntries(args...)
+@deprecate getDataEntryElement(args...) getDataEntryBlob(args...)
```
Comment (Affie) [inline src/Deprecated.jl:49]: I think it's ok to remove all "BigData" references now

File: src/Deprecated.jl (hunk 2 above)
Comment (dehann) [reply src/Deprecated.jl:27]: Could you please help check.  Agree that these should not be in v0.10.

File: src/Deprecated.jl (hunk 3 above)
Comment (dehann) [reply src/Deprecated.jl:15]: I think we should make the promotion of DateTime to include timezone information permanent, not just a short term deprecated feature.

File: src/Deprecated.jl (hunk 4 above)
Comment (dehann) [reply src/Deprecated.jl:49]: Sure, but this style of deprecation does not preserve the sequence of arguments of older API.  This deprecation only preserves the older names but still requires a change in the arguments to follow the new sequence.  This was the origin of #579

File: src/Deprecated.jl (hunk 3 above)
Comment (Affie) [reply src/Deprecated.jl:15]: The promotion is type piracy. So we can't leave it permanently.

File: src/Deprecated.jl (hunk 2 above)
Comment (Affie) [reply src/Deprecated.jl]: Ok, I removed them permanently (again). I couldn't find references in downstream packages.

File: src/Deprecated.jl (hunk 4 above)
Comment (Affie) [reply src/Deprecated.jl]: Removed completely. So not "Big"Data function exist any more.

**Pull request merged**